dask_ml/decomposition/pca.py

```diff
@@ -285,35 +303,53 @@ def _fit(self, X):
         else:
             noise_variance = 0.0
 
-        (
-            self.n_samples_,
-            self.n_features_,
-            self.n_components_,
-            self.components_,
-            self.explained_variance_,
-            self.explained_variance_ratio_,
-            self.singular_values_,
-            self.noise_variance_,
-            self.singular_values_,
-        ) = compute(
-            n_samples,
-            n_features,
-            n_components,
-            components,
-            explained_variance,
-            explained_variance_ratio,
-            singular_values,
-            noise_variance,
-            singular_values,
-        )
-
-        if solver != "randomized":
-            self.components_ = self.components_[:n_components]
-            self.explained_variance_ = self.explained_variance_[:n_components]
-            self.explained_variance_ratio_ = self.explained_variance_ratio_[
-                :n_components
-            ]
-            self.singular_values_ = self.singular_values_[:n_components]
+        try:
+            (
+                self.n_samples_,
+                self.n_features_,
+                self.n_components_,
+                self.components_,
+                self.explained_variance_,
+                self.explained_variance_ratio_,
+                self.singular_values_,
+                self.noise_variance_,
+                self.singular_values_,
+            ) = compute(
+                n_samples,
+                n_features,
+                n_components,
+                components,
+                explained_variance,
+                explained_variance_ratio,
+                singular_values,
+                noise_variance,
+                singular_values,
+            )
+        except ValueError as e:
+            if np.isnan([n_samples, n_features]).any():
+                msg = (
+                    "Computation of the SVD raised an error. It is possible "
+                    "n_components is too large. i.e., "
+                    "`n_components > np.nanmin(X.shape) = "
+                    "np.nanmin({})`\n\n"
+                    "A possible resolution to this error is to ensure that "
+                    "n_components <= min(n_samples, n_features)"
+                )
+                raise ValueError(msg.format(X.shape)) from e
+            raise e
+
+        self.components_ = self.components_[:n_components]
+        self.explained_variance_ = self.explained_variance_[:n_components]
+        self.explained_variance_ratio_ = self.explained_variance_ratio_[:n_components]
+        self.singular_values_ = self.singular_values_[:n_components]
+
+        if len(self.singular_values_) < n_components:
```

Comment:
In scikit-learn, is it possible to get into this state? If so, what do they do?

I suspect they can't get here, because the would have previously raised an exception before the fit. If so, that's what I'd like to do here (after the fit).

Reply:
> n scikit-learn, is it possible to get into this state? If so, what do they do?

No, because the array shape is known ahead of time. I'll raise an exception instead of warning.